Shubham_Sharma/Tic-Tac-Toe/Assets/js/Main.js

```diff
@@ -0,0 +1,117 @@
+var options = "X";
+//Score references
+var p1s = document.getElementById("pl1s");
+var p2s = document.getElementById("pl2s");
+//counters for increasing scores
+var count1 = 0;
+var count2 = 0;
+
+//Getting game button's reference
+var b1 = document.getElementById("b1");
+var b2 = document.getElementById("b2");
+var b3 = document.getElementById("b3");
+var b4 = document.getElementById("b4");
+var b5 = document.getElementById("b5");
+var b6 = document.getElementById("b6");
+var b7 = document.getElementById("b7");
+var b8 = document.getElementById("b8");
+var b9 = document.getElementById("b9");
+var btns = [b1, b2, b3, b4, b5, b6, b7, b8, b9];
+
+//Resets every button's text and color
+function reset(){
+    for(let i=0; i<9; i++){
+        btns[i].innerHTML = "";
+        btns[i].style.backgroundColor = "#a52a2a"
+        btns[i].disabled = false;
+    }
+}
+
+//Checks for any WINNING pattern, and will call a method for highlighting those button's color
+function checker(){
+    var b1 = document.getElementById("b1").innerHTML;
+    var b2 = document.getElementById("b2").innerHTML;
+    var b3 = document.getElementById("b3").innerHTML;
+    var b4 = document.getElementById("b4").innerHTML;
+    var b5 = document.getElementById("b5").innerHTML;
+    var b6 = document.getElementById("b6").innerHTML;
+    var b7 = document.getElementById("b7").innerHTML;
+    var b8 = document.getElementById("b8").innerHTML;
+    var b9 = document.getElementById("b9").innerHTML;
+    
+    (((b1=="X") || (b1=="O")) && ((b1 == b2) && (b2 == b3))) ? highLight("b1", "b2", "b3", b1)
```

Comment:
still violation of DRY

suggestion: ternary operator should be used up to 3 levels is good if more than max 3 levels, you should change your approach

Reply:
Resolved the violation of DRY. Kindly check and let me know if anything else is there.